src/Compilers/CSharp/Test/Emit/Attributes/InternalsVisibleToAndStrongNameTests.cs

```diff
@@ -1185,27 +1201,45 @@ public void SignModuleKeyFileAttr()
             var x = s_keyPairFile;
             string s = String.Format("{0}{1}{2}", @"[assembly: System.Reflection.AssemblyKeyFile(@""", x, @""")] public class C {}");
 
-            var other = CreateStandardCompilation(s, options: TestOptions.ReleaseModule.WithStrongNameProvider(s_defaultProvider));
+            var options = TestOptions.ReleaseModule.WithStrongNameProvider(s_defaultPortableProvider);
+            var other = CreateStandardCompilation(s, options: options);
+
+            var outStrm = new MemoryStream();
+            var success = other.Emit(outStrm);
+            Assert.True(success.Success);
+
+            ConfirmModuleAttributePresentAndAddingToAssemblyResultsInSignedOutput(outStrm, AttributeDescription.AssemblyKeyFileAttribute, legacyStrongName: false);
+        }
+
+        [Fact]
+        public void SignModuleKeyFileAttr_Legacy()
+        {
+            var x = s_keyPairFile;
+            string s = String.Format("{0}{1}{2}", @"[assembly: System.Reflection.AssemblyKeyFile(@""", x, @""")] public class C {}");
+
+            var options = TestOptions.ReleaseModule.WithStrongNameProvider(s_defaultDesktopProvider);
+            var other = CreateStandardCompilation(s, options: options);
 
             var outStrm = new MemoryStream();
             var success = other.Emit(outStrm);
             Assert.True(success.Success);
```

Comment:
Consider simplifying to `var outStrm = other.EmitToStream(expectedDiagnostics);` in these tests.

Reply:
Done

Reply:
Note: a bunch of these have been changed to also emit metadata-only assemblies.